intake_hive/hive_source.py

```diff
@@ -0,0 +1,208 @@
+import threading
+from typing import Dict, List
+from urllib.parse import ParseResult, urlparse, parse_qs  # noqa: F401
+
+import pandas as pd
+import numpy as np
+import pkg_resources
+from pyspark import sql, SparkContext
+from pyspark.sql import SparkSession
+from pyspark.sql.types import (
+        ByteType,
+        ShortType,
+        IntegerType,
+        FloatType,
+        LongType,
+        DoubleType,
+        BooleanType,
+        TimestampType,
+    )
+from intake import DataSource, Schema
+
+
+class SparkHolder(object):
+    session: SparkSession = None
+    lock = threading.Lock()
+
+    def set_session(self, session=None) -> SparkSession:
+        """Set global SQL SparkSession
+        Args:
+            session: Explicitly provide a session object, if you have one
+        """
+        if self.session is not None:
+            return self.session
+
+        with self.lock:
+            if session is None:
+                self.session = sql.SparkSession.builder.enableHiveSupport().getOrCreate()
+            else:
+                self.session = session
+
+        return self.session
+
+    def setup(self) -> SparkSession:
+        return self.set_session()
+
+    def __enter__(self):
+        return self.setup()
+
+    def __exit__(self, *exc):
+        sc: SparkContext = self.session.sparkContext
+        sc.stop()
```

Comment:
stop session before context? Not sure if it will make a diff in the real world but just in case

Reply:
no longer there :)